types/part_set.go

```diff
@@ -270,9 +270,12 @@ func (ps *PartSet) Total() uint32 {
 }
 
 func (ps *PartSet) AddPart(part *Part) (bool, error) {
+	// TODO: remove this? would be preferable if this only returned (false, nil)
```

Comment:
Essentially all `PartSet` methods handle this condition in which the block set is not initialized. This might make the consensus logic simpler, but it is not really easy to read.

Reply:
I believe AddPart is only called in one place in line 1934 in consensus/state and we check right before if the parts are nil, so it should be totally redundant to have the check in here, but we thought it might break tests or something